### ISSUE TITLE
ldif-diff: Repeated changeType entries in output file

### ISSUE THREAD

**Issue by lbascones-dg** [pingidentity/ldapsdk#108]
Dear sirs,

I ran into an issue that I would consider rather serious (unless it is user error, of course). Perhaps there is an explanation, even though I can't think of what it could be. Here's a quick description:

I pulled a list of OUs from AD using ldifde to generate an LDIF file, then modified the OU tree, and finally pulled an updated list of OUs into another LDIF file. I took these two LDIF files and ran LDIF-DIFF on them. I expected to see the list of operations to add the newly added OUs, which is exactly what I saw. The problem is that the **changetype** operations appear twice for every new OU that was added (see below for an example). 

The originating LDIF files (before and after) look fine.

Is this a known issue? If so, are there any plans to address it?

Thanks much!

-LuisB

#########  Segment of LDIF file generates by LDIF-DIFF ##################

# The following entry was in the target LDIF file but not the source LDIF file:
dn: OU=new5,OU=TestBaseOU,DC=ldaptest,DC=com
changetype: add
changetype: add
objectClass: top
objectClass: organizationalUnit
ou: new5

# The following entry was in the target LDIF file but not the source LDIF file:
dn: OU=AnotherNewOrgUnit,OU=NewOrgUnit,OU=TestBaseOU,DC=ldaptest,DC=com
changetype: add
changetype: add
objectClass: top
objectClass: organizationalUnit
ou: AnotherNewOrgUnit

# The following entry was in the target LDIF file but not the source LDIF file:
dn: OU=new2,OU=NewOrgUnit,OU=TestBaseOU,DC=ldaptest,DC=com
changetype: add
changetype: add
objectClass: top
objectClass: organizationalUnit
ou: new2

**Comment by dirmgr** [comment]
The input files for the ldif-diff utility are expected to be files containing LDIF entries, but my guess is that the file you have provided contains LDIF change records rather than entries. But because ldif-diff expects the input to be entries, then when it sees "changetype: add", it interprets that as an attribute named "changetype" with a value of "add".

On the other hand, the output that ldif-diff is in the form of change records rather than entries. That means that each of those records will have a "changetype" element that indicates what type of change you need for an entry between the source and target LDIF files. So in your output, the first changetype value is the actual changetype, and the second is what ldif-diff thinks is an attribute called "changetype".

If you strip the "changetype" lines out of our source and target LDIF files, then that should resolve the problem.

**Comment by lbascones-dg** [comment]
You are absolutely correct. The output generated by ldifde seems to contain change records rather than plain entries. Unfortunately, there seems to be no way to modify this behavior. I may have to find a different way to extract contents into an LDIF file.

Thanks for the heelp,

-LuisB
